src/modules/Game.class.js

```diff
@@ -20,25 +20,259 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.state = Game.copyState(initialState);
+    this.initialState = Game.copyState(initialState);
+    this.score = 0;
+    this.status = 'idle';
   }
+  static copyState(state) {
+    return state.map((row) => row.slice());
+  }
+
+  static moveRowLeft(row, game) {
+    let filteredRow = row.filter((value) => value !== 0);
+
+    for (let i = 0; i < filteredRow.length - 1; i++) {
+      if (filteredRow[i] === filteredRow[i + 1]) {
+        filteredRow[i] *= 2;
+        filteredRow[i + 1] = 0;
+        game.score += filteredRow[i];
+
+        if (filteredRow[i] === 2048) {
+          game.status = 'win';
+        }
+      }
+    }
+    filteredRow = filteredRow.filter((value) => value !== 0);
+
+    while (filteredRow.length < 4) {
+      filteredRow.push(0);
+    }
+
+    return filteredRow;
+  }
+
+  moveLeft() {
+    if (this.status === 'playing') {
+      let changed = false;
+
+      for (let r = 0; r < 4; r++) {
+        const originalRow = [...this.state[r]];
+
+        this.state[r] = Game.moveRowLeft(this.state[r], this);
+
+        if (!changed && originalRow.toString() !== this.state[r].toString()) {
+          changed = true;
+        }
+      }
+
+      if (changed) {
+        this.addRandomTile();
+
+        if (!this.canMove()) {
+          this.status = 'lose';
+        }
+      }
+    }
+  }
+  static moveRowRigth(row, game) {
+    let filteredRow = row.filter((value) => value !== 0);
+
+    for (let i = filteredRow.length - 1; i > 0; i--) {
+      if (filteredRow[i] === filteredRow[i - 1]) {
+        filteredRow[i] *= 2;
+        filteredRow[i - 1] = 0;
+        game.score += filteredRow[i];
+
+        if (filteredRow[i] === 2048) {
+          game.status = 'win';
+        }
+      }
+    }
+
+    filteredRow = filteredRow.filter((value) => value !== 0);
+
+    while (filteredRow.length < 4) {
+      filteredRow.unshift(0);
+    }
+
+    return filteredRow;
+  }
+  moveRight() {
+    if (this.status === 'playing') {
+      let changed = false;
+
+      for (let r = 0; r < 4; r++) {
+        const originalRow = [...this.state[r]];
```

Comment:
in moveRight and moveLeft you duplicate almost all of the code except the method name you call inside (Game.moveRowRigth). Create a method `move` where you will do this thing and which will call the correct method based on the passed direction as an argument ('left' | 'right', etc.)